### ISSUE TITLE
JSON TimelineJS Eras in Timeline JS3

### ISSUE THREAD

**Issue by kcyarn** [NUKnightLab/TimelineJS3#189]
A few years ago, I created a WordPress plugin that uses a custom post type for the individual events. This lets me use a locally hosted WordPress instance to create and edit my timeline without hand coding json or making my data public. Now, I'm converting over to TimelineJS3. 

Compared to the previous script, JS3 flies. It didn't seem to notice when I added my 137th event.

That said, I'm confused about how TimelineJS's eras fit into TimelineJS3's json.

Additionally, convert_json.py shows a different example json than the example json provided in README.md. Which is correct?

**Comment by kcyarn** [comment]
So I answered my own question.

```
 "era": [
            {
                "startDate":"2011,12,10",
                "endDate":"2011,12,11",
                "headline":"Headline Goes Here",
                "tag":"This is Optional"
            }
        ]
```
from Timeline JS does not appear to exist as such in Timeline JS3.

The first JS3 Wordpress plugin attempt simply pulled the custom era event types into the json as if they events. It works.

Logically, there is no reason not to do this. However, promoting the eras from below the timenav into the timeline itself made my timeline extremely cluttered. 137 events when I last posted and it may have grown a little since then...Cluttered's almost a given at this point.

In the end, I created a category for each era then used an sql insert statement to add the category if the start date was between the era dates. A few tweaks to the shortcode and json.php and I can now filter the events by category, which I probably should have done from the beginning.

In the off chance that anyone else needs to do this or something similar, I pushed the plugin to github. It's not the prettiest code. It hasn't been vetted and probably won't be. (I only use it on a localhost that doesn't even have internet access 90% of the time, so not a lot of incentive to go through the approval process.) 

[timeline-js3-wp](https://github.com/kcyarn/timeline-js3-wp)

**Comment by JoeGermuska** [comment]
@zachwise I can't remember where we left off with implementing 'eras' -- it's still work-to-be-done, right?

**Comment by xabispacebiker** [comment]
I am also interested on the implementation of eras in Timeline JS3. It was a cool feature. I haven't checked TimelineJS3 yet, (so I did with TimelineJS) then .. Do you think it should be possible/easy to implement the feature following the TimelineJS as an example or JS3 has changed that much? I could give a hand on this if needed.

(thanks a lot for this amazing timeline, the best i've seen so far.)

**Comment by Ichag** [comment]
+1

**Comment by zachwise** [comment]
Closed via 5d38316cd9cd6624e564abb6f834381c7b7a3057
Example JSON with eras: https://github.com/NUKnightLab/TimelineJS3/blob/master/compiled/examples/eras.json
